ambari-server/src/main/java/org/apache/ambari/server/state/ServiceImpl.java

```diff
@@ -713,20 +720,16 @@ public boolean isSsoIntegrationDesired() {
   }
 
   public boolean isSsoIntegrationEnabled() {
-    return ssoIntegrationSupported && ssoEnabledConfigValid() && "true".equalsIgnoreCase(ssoEnabledConfigValue());
-  }
-
-  private boolean ssoEnabledConfigValid() {
-    return ssoEnabledConfiguration != null && ssoEnabledConfiguration.split("/").length == 2;
+    try {
+      final Predicate ssoEnabledPredicate = ssoEnabledConfiguration == null ? null : PredicateUtils.fromJSON(ssoEnabledConfiguration);
```

Comment:
This does not seem to be backwards compatible with the existing implementation.  If it is not, there may be issues with existing mpacks that support this feature. 

I suggest using a different property for the predicate declaration so that old way can be deprecated and write a message to the log if encountered.

Reply:
ok; will submit the change soon